src/fmu/dataio/providers/_fmu.py

```diff
@@ -248,26 +250,36 @@ def _get_restart_data_uuid(self) -> UUID | None:
         """Load restart_from information"""
         assert self._runpath is not None
         logger.debug("Detected a restart run from environment variable")
-        restart_path = self._runpath / os.environ[RESTART_PATH_ENVNAME]
-        restart_case_metafile = (
-            restart_path.parent.parent / ERT_RELATIVE_CASE_METADATA_FILE
-        ).resolve()
+        restart_path = (self._runpath / os.environ[RESTART_PATH_ENVNAME]).resolve()
 
-        if not restart_case_metafile.exists():
+        if _casepath_has_metadata(restart_path.parent.parent):
```

Comment:
ATM we do not support runs without iterations folder (we use `parent.parent),` will make separate issue of it